### ISSUE TITLE
Question: Is it possible to jump when current text is still selected?

### ISSUE THREAD

**Issue by tjdevries** [hrsh7th/vim-vsnip#48]
I'm trying to figure out how to do something using the example snippet linked in the readme:

```
{
  "Class": {
    "prefix": ["class"],
    "body": [
      "class $1 ${2:extends ${3:Parent} }{",
      "\tconstructor() {",
      "\t\t$0",
      "\t}",
      "}"
    ],
    "description": "Class definition template."
  }
}
```

What I'd like to do is be able to jump past "extends" when it hops to there after the first expand.

Maybe a little more clear:

Step 1:
```
class
```

Step 2: Expand

```
class | extends Parent {
    constructor() {
        
    }
}
```

Step 3: Type in name:

```
class MyClass extends Parent {
    constructor() {
        
    }
}
```

Step 4: Jump to next

```
class MyClass |extends Parent| {
    constructor() {
        
    }
}
```

(extends parent is highlighted)

My question is how to jump to the next placeholder without changing or typing anything here (for me, it does not seem that `vsnip-jump-next` works the way I expect).

In other words I'd like to be able to cycle forward and backward on the placeholders without typing anything other than "jump-next".

**Comment by hrsh7th** [comment]
Your issue seems caused by failed to map to `smap` for me.

Could you provide your configuration?

**Comment by tjdevries** [comment]
Oh, you're right -- I didn't understand what the point of doing the `smap`s were in the config examples. Once I did that it worked fine! Thank you :D

**Comment by hrsh7th** [comment]
Thank you for interest to vsnip!
